internal/gatewayapi/route.go

```diff
@@ -390,16 +390,40 @@ func (t *Translator) processGRPCRouteRule(grpcRoute *GRPCRouteContext, ruleIdx i
 		}
 
 		if match.Method != nil {
-			if match.Method.Method != nil {
-				irRoute.HeaderMatches = append(irRoute.HeaderMatches, &ir.StringMatch{
-					Name:  ":method",
-					Exact: match.Method.Method,
-				})
-			}
-			/* TODO
-			if match.Method.Service != nil {
+			// GRPC's path is in the form of "/<service>/<method>"
+			//nolint:gocritic
```

Comment:
why was the nolint needed ?

Reply:
add nolint to ignore 'Error: ifElseChain: rewrite if-else to switch statement (gocritic)'